### DISCUSSION TITLE
Second service added to wsdl file

### DISCUSSION THREAD

**Discussion by fab1aan** [quarkiverse/quarkus-cxf#450] [Q&A]
I'm trying to migrate a webservice to quarkus. I specify the wsdl file as follows in application.properties:

quarkus.cxf.endpoint."/asdfService".wsdl=/wsdl/asdfService.wsdl

In asdfService.wsdl, I provide the following service:

```
<wsdl:service name="asdfService">
	<wsdl:port binding="tns:asdfServiceBinding" name="asdfServicePort">
		<soap:address location="urn:DynamicEndpoint"/>
	</wsdl:port>
</wsdl:service>
```

Now, localhost:8080/asdfService?wsdl returns the wsdl but there are 2 services:

```
<wsdl:service name="asdfPortTypeService">
   <wsdl:port binding="tns:asdfPortTypeSoapBinding" name="asdfPortTypePort">
   <soap:address location="http://localhost:8080/asdfService"/>
   </wsdl:port>
</wsdl:service>
<wsdl:service name="asdfService">
   <wsdl:port binding="tns:asdfServiceBinding" name="asdfServicePort">
   <soap:address location="urn:DynamicEndpoint"/>
   </wsdl:port>
</wsdl:service>

```
The second service is taken from the provided wsdl file, but the first one seems to be added by quarkus-cxf. What could be the reason for that? Also, urn:DynamicEndpoint should be replaced by http://localhost:8080/asdfService.  Thanks for your help!

**Comment by shumonsharif** [comment]
For the multiple services, you may have a mismatch between your `@WebService` annotations in your interface and implementation classes.  Hard to tell though; are you able to provide a reproducer?

Regarding the 'urn:DynamicEndpoint' it seems as though you may be mixing code first / contract first approaches ... again, a reproducer would be great to provide any suggestions!

**Reply by fab1aan** [reply]
It's supposed to be contract first. I have an intergration-project (external dependency) which contains the wsdl, generated service endpoint interface and generated classes from the types defined in the wsdl. In my project I have the implementation of the service endpoint interface (annotated with @WebService). The weird thing is, I only get it to work if I copy paste the service endpoint interface into my own project - it doesn't work at all if I import it from the dependency (then endpoint returns "Resource not found")

**Reply by shumonsharif** [reply]
Sounds like you may be missing a jandex index.  See this link for details:. https://github.com/quarkiverse/quarkus-cxf/discussions/365 and keep me posted in case that doesn't work.

**Reply by fab1aan** [reply]
Yess that solved it! application.properties:

```
quarkus.index-dependency.endpointinterfaceClass.group-id=asdf.integration
quarkus.index-dependency.endpointinterfaceClass.artifact-id=asdf-integration
```

One problem down, now I need to figure out why there's 2 services. Thanks again for your help!

**Reply by shumonsharif** [reply]
You may possibly not need to define this property, can. you remove it and try?  
```quarkus.cxf.endpoint."/asdfService".wsdl=/wsdl/asdfService.wsdl```

Also make sure you define something like:
```quarkus.cxf.endpoint."/asdfService".implementor=org.acme.AsdfServiceImpl```

**Reply by fab1aan** [reply]
Sorry for the late reply, I was away last week. If I remove the .wsdl property, then the wsdl is not 100% right anymore. For example, some elements with type decimal are shown as strings. It seems like the wsdl gets generated somehow instead of taking the wsdl file from the integration project. But you're right, if I remove it then there is only 1 service defined. Also, I noticed that the wsdl:binding element is duplicated as well, but both reference the right port type (wsdl:portType). It looks like the framework tries to add a service to the wsdl instead to use the service & binding defined in the given wsdl file (provided with quarkus.cxf.endpoint."/asdfService".wsdl=/wsdl/asdfService.wsdl).

**Reply by fab1aan** [reply]
I changed the name of the service (in the wsdl file) from AsdfService to AsdfPortTypeService and now it seems to work:

`<wsdl:service name="AsdfPortTypeService">
 <wsdl:port binding="tns:AsdfServiceBinding" name="AsdfServicePort">
  <soap:address location="urn:DynamicEndpoint"/>
 </wsdl:port>
</wsdl:service>`

My service endpoint interface is named AsdfPortType => is there a convention that the service must be named the same with the ending "Service"?

It's not a really nice solution, because now I need to change the integration project. Any ideas what went wrong or explanations of this behavior? Thanks!

**Reply by shumonsharif** [reply]
The behavior you are seeing may be related to this issue:  https://github.com/quarkiverse/quarkus-cxf/issues/439 which I plan on including in the next release (likely in the next week or so).
